src/wui/interactive_player.cc

```diff
@@ -342,7 +342,14 @@ void InteractivePlayer::think() {
 					set_sel_pos(Widelands::NodeAndTriangle<>{
 					   flag_to_connect_,
 					   Widelands::TCoords<>(flag_to_connect_, Widelands::TriangleIndex::D)});
-					start_build_road(flag_to_connect_, field.get_owned_by());
+					const Widelands::Map& map = egbase().map();
+					if (map.get_waterway_max_length() >= 2 &&
+					    Widelands::CheckStepFerry(egbase()).reachable_dest(
+					       map, Widelands::FCoords(flag_to_connect_, &field))) {
+						show_field_action(this, get_player(), &fieldaction_);
+					} else {
+						start_build_road(flag_to_connect_, field.get_owned_by());
+					}
```

Comment:
Found it. During merging you replaced this line:```start_build_road(flag_to_connect_, field.get_owned_by());``` with the old code.

Reply:
Ok fixed, well spotted.
I had stacked this branch on the ferry branch on LP, so I got masses of merge conflicts now… :)